pastas/model.py

```diff
@@ -480,12 +480,16 @@ def residuals(
             tmax = self.settings["tmax"]
         if freq is None:
             freq = self.settings["freq"]
+        if self.settings["freq_obs"] is None:
+            freq_obs = freq
+        else:
+            freq_obs = self.settings["freq_obs"]
```

Comment:
```suggestion
        freq_obs = self.settings.get("freq_obs", freq)
```

Reply:
Clearly won't work. settings does not have a get. Author appears to be pushing his GH profile.

Reply:
Well, it does work: settings is a dictionary, and a dictionary has a get-method. However, when `settings["freq_obs"]` is `None`, it will not revert to `freq`. It only does when `"freq_obs"` is not in `self.settings`. So this change will not perform as desired. We can shorten the code by:
`freq_obs  = self.settings["freq_obs"] if self.settings["freq_obs"] is not None else freq`

Reply:
Ah ok because the key `freq_obs` is always present but with None as value right?

i see i see :-) my bad ;-)